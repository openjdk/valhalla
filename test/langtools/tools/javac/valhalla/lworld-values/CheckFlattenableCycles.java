/*
 * @test /nodynamiccopyright/
 * @summary Check for cycles through fields declared flattenable.
 * @compile -XDallowValueMemberCycles -XDrawDiagnostics CheckFlattenableCycles.java
 * @compile/fail/ref=CheckFlattenableCycles.out -XDrawDiagnostics CheckFlattenableCycles.java
 */

final primitive class CheckFlattenableCycles {
    class InnerRef {
        CheckFlattenableCycles cfc;
    }
    primitive final class InnerValue {
        final CheckFlattenableCycles     cfc = CheckFlattenableCycles.default; // Error.
    }
    final CheckFlattenableCycles cfc = CheckFlattenableCycles.default; // Error.
    final int i = 10;
    final String s = "blah";
    final InnerRef ir = new InnerRef(); // OK.
    final InnerValue     iv = InnerValue.default; // Error
}
