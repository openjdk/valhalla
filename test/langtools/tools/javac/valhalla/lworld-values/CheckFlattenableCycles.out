CheckFlattenableCycles.java:13:42: compiler.err.cyclic.primitive.class.membership: CheckFlattenableCycles.InnerValue
CheckFlattenableCycles.java:15:34: compiler.err.cyclic.primitive.class.membership: CheckFlattenableCycles
CheckFlattenableCycles.java:19:26: compiler.err.cyclic.primitive.class.membership: CheckFlattenableCycles
3 errors
